src/include/storage/storage_structure/disk_array.h

```diff
@@ -116,6 +116,10 @@ class BaseDiskArray {
     // The return value is the idx of val in array.
     uint64_t pushBack(U val);
 
+    // Note: This function is to be used only by the WRITE trx. Currently, this function doesn't
+    // support shrink the size of the array.
```

Comment:
This is another example where how the addressing was done is not clear.

Reply:
Removed the part on write transaction. See this the new one: 
```
    // Note: Currently, this function doesn't support shrinking the size of the array.
```

I guess it might because i force pushed. I will try to not force push next time, see if github can correctly show changes since your last review. That might come more handy.